### PR TITLE
[RFR] Refactor project collection/entity methods/properties

### PR DESCRIPTION
**Pull request by ganeshhubale**
Signed-off-by: Ganesh Hubale <ghubale@redhat.com>
- Need this PR https://github.com/windup/windup_integration_test/pull/46 to be merged.

### CODE REVIEW

File: mta/entities/__init__.py [hunk 1]
```diff
@@ -68,26 +69,40 @@ class AllProjectView(BaseLoggedInPage):
...
-        remove = Text(".//div[contains(@class, 'no-matches')]/a")
+        text = Text(".//div[contains(@class, 'pf-c-empty-state__body')]")
 
     def clear_search(self):
         """Clear search"""
         if self.search.value:
             self.search.fill("")
 
+    @property
+    def all_projects(self):
```
Comment (mayurilahane) [inline mta/entities/__init__.py:99]: I guess this property name can be more generic as "Clear_filters".
Thanks!

Comment (ganeshhubale) [reply mta/entities/__init__.py:99]: Nice catch :+1:

File: mta/entities/projects.py [hunk 2]
```diff
@@ -306,7 +306,10 @@ class Project(BaseEntity, Updateable):
     def exists(self):
         """Check project exist or not"""
         view = navigate_to(self.parent, "All")
-        return view.projects.exists(self.name)
+        for row in view.table:
```
Comment (sshveta) [inline mta/entities/projects.py:309]: You can do if 

             if self.get_project(name):
                  return True
             return False

**Review by sshveta — CHANGES_REQUESTED** [review]

File: mta/entities/projects.py (hunk 2 above)
Comment (ganeshhubale) [reply mta/entities/projects.py:309]: @sshveta it is not possible. `get_project` is a method of collection not entity.
```
In [1]: col = app.collections.projects

In [2]: y = col.all()[1]

In [3]: y.exists
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-3-82864ce5a938> in <module>
----> 1 y.exists

~/migration_toolkit_for_applications/windup_integration_test/mta/entities/projects.py in exists(self)
    311         #         return True
    312         # return False
--> 313         if self.get_project(self.name):
    314             return True
    315         return False

AttributeError: 'Project' object has no attribute 'get_project'
```

**Review by digitronik — APPROVED** [review]

**Pull request merged**